pkg/slack/client.go

```diff
@@ -236,6 +236,54 @@ func (c *Client) GetUserGroups(ctx context.Context, teamID string) ([]slack.User
 	return res.UserGroups, nil
 }
 
+// SetWorkspaceRole sets the role for the given user in the given team.
+func (c *Client) SetWorkspaceRole(ctx context.Context, teamID, userID, roleID string) error {
+	values := url.Values{
+		"token":   {c.token},
+		"team_id": {teamID},
+		"user_id": {userID},
+	}
+
+	var role string
+
+	if roleID != "" {
+		roleSplit := strings.Split(roleID, ":")
+		if len(roleSplit) >= 2 {
+			role = roleSplit[1]
+		}
+	}
+
+	var action string
+	switch role {
+	case "owner":
+		action = "setOwner"
+	case "admin":
+		action = "setAdmin"
+	case "":
+		action = "setRegular"
+	default:
+		return fmt.Errorf("invalid role type: %s", role)
+	}
+
+	userGroupsUrl, err := url.JoinPath(baseUrl + "admin.users." + action)
```

Comment:
I think we want a comma instead of a +

```suggestion
	userGroupsUrl, err := url.JoinPath(baseUrl, "admin.users." + action)
```

Reply:
I will merge this but the fix will be in the PR for SCIM api. I really don't know why i put + here :D